libs/sim/storage.rs

```diff
@@ -51,22 +65,90 @@ impl SimElement for Storage {
     }
 }
 
+fn load_lib(plugin_str: &str) -> libloading::Library {
```

Comment:
This should not be copied. Instead make load_lib a generally accessible util function.

Reply:
Yeah that's a good call.  I made it its own module.

Reply:
Ah I would recommend making a utils.rs module instead and putting all general purpose functions we use in there. This will grow in time.

Reply:
Ah I realized we are dealing with the simulator. Might not be necessary there. Instead I would just make it publicly accessible  from simulator.rs for example. Just need to find a good place for it.

Reply:
I think it should be a part of the "bare bones" simulator code, so it should go in libs/sim.  From there it should be available to anything in that crate and, because bookinfo inherits from that, anything in bookinfo or any other environment we make.